main.ts

```diff
@@ -18,12 +19,18 @@ import {
 
 interface ShareAsGistSettings {
   enableUpdatingGistsAfterCreation: boolean;
+  enableAutoSaving: boolean;
+  showAutoSaveNotice: boolean;
+  autoSaveDelaySeconds: number;
   includeFrontMatter: boolean;
 }
 
 const DEFAULT_SETTINGS: ShareAsGistSettings = {
   includeFrontMatter: false,
   enableUpdatingGistsAfterCreation: true,
+  enableAutoSaving: true,
+  autoSaveDelaySeconds: 10,
```

Comment:
What's the thinking behind this delay? Can you explain how it fits into the UX?

Reply:
It's basically a debounce, without the delay the document would be synced very often and the GitHub token would be rate limited. This is mainly the configuration option I was talking about when I said I'd added more configuration then necessary, this could just be hardcoded to a sensible value.

Reply:
I think I'd probably just hard-code it to a sensible value. Let's go for somewhere between 10 and 30 seconds, I reckon.